### PR TITLE
UI: Part 2 - hds adoption replace <Modal> 

### PR DESCRIPTION
**Pull request by hellobontempo**
transit key action modals

### CODE REVIEW

File: ui/app/templates/components/transit-key-action/datakey.hbs [hunk 1]
```diff
@@ -74,19 +74,27 @@
     </div>
   </div>
 </form>
-<Modal @title="Copy your generated key" @onClose={{action (mut @isModalActive) false}} @isActive={{@isModalActive}}>
-  <section class="modal-card-body">
-    <div class="box is-shadowless is-fullwidth is-sideless">
+{{#if @isModalActive}}
```
Comment (hellobontempo) [inline ui/app/templates/components/transit-key-action/datakey.hbs:77]: Screenshot 2023-09-28 at 1 55 20 PM

File: ui/app/templates/components/transit-key-action/verify.hbs [hunk 2]
```diff
@@ -192,16 +192,25 @@
     </div>
   </div>
 </form>
-<Modal @title="Results" @onClose={{action (mut @isModalActive) false}} @isActive={{@isModalActive}}>
```
Comment (hellobontempo) [inline ui/app/templates/components/transit-key-action/verify.hbs:195]: Previously the modal didn't even have a close button! 🤯 
Screenshot 2023-09-28 at 2 05 35 PM
Screenshot 2023-09-28 at 2 05 49 PM

File: ui/app/templates/components/transit-key-action/verify.hbs [hunk 3]
```diff
@@ -192,16 +192,25 @@
...
-        The input is
-        {{if @valid "valid" "not valid"}}
-        for the given
-        {{if @signature "signature." "HMAC."}}
-      </A.Description>
-    </Hds::Alert>
-  </section>
-</Modal>
\ No newline at end of file
+{{#if @isModalActive}}
```
Comment (hellobontempo) [inline ui/app/templates/components/transit-key-action/verify.hbs:195]: Screenshot 2023-09-28 at 12 59 22 PM
Screenshot 2023-09-28 at 2 05 04 PM

File: ui/package.json [hunk 4]
```diff
@@ -65,7 +65,8 @@
     "@ember/test-waiters": "^3.0.0",
     "@glimmer/component": "^1.1.2",
     "@glimmer/tracking": "^1.1.2",
-    "@hashicorp/ember-flight-icons": "^3.0.9",
+    "@hashicorp/design-system-components": "^2.12.2",
```
Comment (hellobontempo) [inline ui/package.json:68]: had to upgrade because `@container` support for `<Hds::Copy::Snippet>` was added recently

**Review by kiannaquach — APPROVED** [review]
🎉 double checked the `@container` arg for the `CopyButton`s! looks good~ might be good to get a 2nd set of 👀 just in case!

**Pull request merged**